platform/ios/src/MGLMapView.mm

```diff
@@ -1169,6 +1169,10 @@ - (void)touchesBegan:(__unused NS_SET_OF(UITouch *) *)touches withEvent:(__unuse
 {
     _changeDelimiterSuppressionDepth = 0;
     _mbglMap->setGestureInProgress(false);
+    if (self.userTrackingState == MGLUserTrackingStateBegan)
+    {
+        [self setUserTrackingMode:MGLUserTrackingModeNone animated:NO];
```

Comment:
That’s a good point when it comes to the [initial transition](https://github.com/mapbox/mapbox-gl-native/blob/13f11635921b660cd4817a1615a003ea183b3005/platform/ios/src/MGLMapView.mm#L4262) for the first user location update. This is definitely an improvement in that case. On the other hand, we also [set `MGLUserTrackingStateBegan`](https://github.com/mapbox/mapbox-gl-native/blob/13f11635921b660cd4817a1615a003ea183b3005/platform/ios/src/MGLMapView.mm#L4227) before animating in response to a significant user location update. Consider the following scenario:

1. The user is in user location tracking mode while a passenger in a moving car.
1. The user enters a tunnel.
1. Upon emerging from the tunnel half a mile away, `-didUpdateLocationSignificantlyAnimated:` is called, setting `MGLUserTrackingStateBegan` before flying to the new location.
1. While flying to the new location, the user double-taps to zoom in.

Double-tapping ideally wouldn’t stop the transition or boot the user out of user tracking mode. However, as you’ve noted, we do call `cancelTransitions()` because we lack the ability to perform animations concurrently (#3625). Still, the current code automatically resumes following the user’s location upon the next location update, whereas this change means the user will have to somehow reenter user location tracking mode.

I wonder if there’s some way to persist the user tracking mode if the current transition is due to a significant user location update, but not if the current transition is the _first_ user location update.

/cc @frederoni